library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -121,6 +117,31 @@ internal class TransactionPayloadFragment : Fragment(), TransactionFragment, Sea
         return true
     }
 
+    private class UiLoaderTask(val fragment: TransactionPayloadFragment) :
+        AsyncTask<Pair<Int, HttpTransaction>, Unit, Triple<String, String, Boolean>>() {
+
+        override fun doInBackground(vararg params: Pair<Int, HttpTransaction>):
+        Triple<String, String, Boolean> {
+            val (type, transaction) = params[0]
+            return when (type) {
+                TYPE_REQUEST -> Triple(
+                    transaction.getRequestHeadersString(true),
+                    transaction.getFormattedRequestBody(),
+                    transaction.isRequestBodyPlainText
+                )
+                else -> Triple(
+                    transaction.getResponseHeadersString(true),
+                    transaction.getFormattedResponseBody(),
+                    transaction.isResponseBodyPlainText
+                )
+            }
+        }
+
+        override fun onPostExecute(result: Triple<String, String, Boolean>) {
```

Comment:
How about having a progress bar and hiding it in the `onPreExecute`?

Reply:
I rebased. I think a progress bar could be nice, but can be done separately.